src/search.js

```diff
@@ -0,0 +1,47 @@
+const checkMatch = (value, searchTerm) => value.toLowerCase().includes(searchTerm);
+
+const filter = (currentNode, searchTerm, onlyPublic) => {
+  let newChildren = [];
+  let isPublic = true;
+  let result = false;
+  let hasChildren;
+
+  currentNode.hasOwnProperty("children") && currentNode.children.forEach(child => {
+    let keepChild = filter(child, searchTerm, onlyPublic);
+
+    if (keepChild) {
+      newChildren.push(child);
+    }
+  });
+
+  if (currentNode.hasOwnProperty("children")) {
+    currentNode.children = newChildren;
+  }
+
+  hasChildren = (currentNode.hasOwnProperty("children") && currentNode.children.length);
+
+  let matched = (
+    searchTerm === '' ||
```

Comment:
One of the things I wanted to try to do but ran out of time for was to pass filters down to the search. I would create filters that have a `call` function: `SearchFilter`, `PublicFilter`, something like that. I was hoping to do something like `filters.forEach(filter => filter.call())` There is some connectedness to the `searchTerm` and `onlyPublic` so this may not have worked as I wanted. But I wanted to make it more extensible, so new filters could be added.